src/marqo/tensor_search/tensor_search.py

```diff
@@ -759,8 +832,44 @@ def _add_documents_structured(config: Config, add_docs_params: AddDocsParams, ma
                                      f"Is this a new tensor field?")
 
                 if len(chunks) == 0:  # Not using existing tensors or didn't find it
-                    is_image_pointer_field = marqo_field.type == FieldType.ImagePointer
-                    if isinstance(field_content, str):
+                    if marqo_field.type in [FieldType.VideoPointer, FieldType.AudioPointer]:
+                        try:
+                            media_chunks = content_repo[field_content]
+                            
+                            if isinstance(content_repo[field_content], s2_inference_errors.S2InferenceError):
+                                raise content_repo[field_content]
+                            for chunk_index, media_chunk in enumerate(media_chunks):
+                                chunk_start = media_chunk['start_time']
+                                chunk_end = media_chunk['end_time']
+                                chunk_id = f"{field}::start_{chunk_start:.2f}::end_{chunk_end:.2f}"
+                                chunks.append(chunk_id)
+                                
+                                vector = s2_inference.vectorise(
+                                    model_name=marqo_index.model.name,
+                                    content=[media_chunk['tensor']],  # Wrap in list as vectorise expects an iterable
+                                    model_properties=marqo_index.model.get_properties(),
+                                    device=add_docs_params.device,
+                                    normalize_embeddings=marqo_index.normalize_embeddings,
+                                    infer=True,
+                                    model_auth=add_docs_params.model_auth,
+                                    modality=Modality.VIDEO if marqo_field.type == FieldType.VideoPointer else Modality.AUDIO
+                                )
+                                embeddings.extend(vector)  # vectorise returns a list of vectors
+                                
+                        except Exception as e:
```

Comment:
Handling generic Exception class. See comment on this above

Reply:
Yep, should be `s2_inference_errors.S2InferenceError as e:` like the others

Reply:
done